src/server/game/AI/SmartScripts/SmartScriptMgr.h

```diff
@@ -1380,39 +1380,38 @@ struct SmartScriptHolder
 
 typedef std::unordered_map<uint32, WayPoint*> WPPath;
 
-typedef std::list<WorldObject*> ObjectList;
+typedef std::vector<WorldObject*> ObjectList;
 
 class ObjectGuidList
 {
-    ObjectList* m_objectList;
-    GuidList* m_guidList;
+    ObjectList m_objectList;
+    GuidList m_guidList;
     WorldObject* m_baseObject;
 
 public:
-    ObjectGuidList(ObjectList* objectList, WorldObject* baseObject)
+    ObjectGuidList(ObjectList objectList, WorldObject* baseObject)
     {
-        ASSERT(objectList != NULL);
+        ASSERT(!objectList.empty());
```

Comment:
what's wrong passing an empty list ? the old code didn't assert in this case, this whole assert should just be removed in the new code